myriad-scheduler/src/main/java/com/ebay/myriad/Main.java

```diff
@@ -190,6 +196,27 @@ private void startNMInstances(Injector injector) {
       }
     }
 
+    /**
+     * Create NMProfile for any configured service
+     * @param injector
+     */
+    private void initJavaBasedTaskConfiguration(Injector injector) {
+      LOGGER.info("Initializing JavaBasedTaskConfiguration");
+      NMProfileManager profileManager = injector.getInstance(NMProfileManager.class);
+
+      Map<String, AuxTaskConfiguration> auxServicesConfigs = 
+          injector.getInstance(MyriadConfiguration.class).getAuxTaskConfigurations();
+      if (auxServicesConfigs != null) {
+        for (Map.Entry<String, AuxTaskConfiguration> entry : auxServicesConfigs.entrySet()) {
+          AuxTaskConfiguration config = entry.getValue();
+          final Double cpu = config.getCpus().or(AuxTaskConfiguration.DEFAULT_CPU);
+          final Double mem = config.getJvmMaxMemoryMB().or(AuxTaskConfiguration.DEFAULT_MEMORY);
+          
+          profileManager.add(new NMProfile(entry.getKey(), cpu, mem));
```

Comment:
I agree we should have an abstraction that describes what's needed to launch a service (cpu/mem/ports/attributes etc). Currently these are described in .yml as:
```yml
nodemanager:
  jvmMaxMemoryMB: 1024
  cpus: 0.2
  cgroups: false
```
and
```yml
  jobhistory:
     taskFactoryImplName: com.ebay.myriad.scheduler.JHSTaskFactoryImpl
     jvmMaxMemoryMB: 1024
     cpus: 1
     ports: [10033, 10020, 19888]
     envSettings: -Dcluster.name.prefix=/mycluster
     taskName: jobhistory
   timelineserver:
     taskFactoryImplName: com.ebay.myriad.scheduler.TMSTaskFactoryImpl
     jvmMaxMemoryMB: 1024
     cpus: 1
     envSettings: -Dcluster.name.prefix=/mycluster2
     taskName: timelineserver	 
```

But, this shouldn't be mixed with NMProfile. We should define a new java class that encapsulates the specifications needed to launch a service, say, ```ServiceSpec``` and have all the services launched by Myriad (NM/JHS/TLS and maybe eco-servers like Hive in the future) use the ```ServiceSpec```. 

Contrast the above with NMProfile. NMProfile comes from the following definitions in .yml.

```yml
profiles:
  zero:  # NMs launched with this profile dynamically obtain cpu/mem from Mesos
    cpu: 0
    mem: 0
  small:
    cpu: 1
    mem: 1100
  medium:
    cpu: 2
    mem: 2048
  large:
    cpu: 4
    mem: 4096
```

The reasons NMProfile can't be (re)used in place of ServiceProfile:
- It describes the additional amounts of cpu/mem/disk NM should consume (in addition to cpu/mem/disk described in NM's ```ServiceSpec```)
- It describes just cpu/mem/disk. It doesn't (and in the future, it won't) describe things like ports/attributes/commands/env variables etc.